File: sandbox-runner/patterns/continue-as-new/python/worker.py
import asyncio

from temporalio.client import Client
from temporalio.worker import Worker

from activities import fetch_batch, process_record
from shared import TASK_QUEUE
from workflows import DataProcessorWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")
    worker = Worker(
        client,
        task_queue=TASK_QUEUE,
        workflows=[DataProcessorWorkflow],
        activities=[fetch_batch, process_record],
    )
    print(f"Worker listening on task queue '{TASK_QUEUE}'", flush=True)
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: sandbox-runner/patterns/continue-as-new/go/worker.go
package main

import (
	"log"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	w := worker.New(c, TaskQueue, worker.Options{})
	w.RegisterWorkflow(DataProcessorWorkflow)
	w.RegisterActivity(FetchBatch)
	w.RegisterActivity(ProcessRecord)

	log.Printf("Worker listening on task queue '%s'", TaskQueue)
	if err := w.Run(worker.InterruptCh()); err != nil {
		log.Fatalln("Worker run failed:", err)
	}
}
